### ISSUE TITLE
[EEE][Crop & Soil] Disentangle EEE from other modules

### ISSUE THREAD

**Issue by ew3361zh** [RuminantFarmSystems/RuFaS#2077]
The connection between the EEE module and other modules should be disentangled so that they meet the requirements outlined in https://github.com/RuminantFarmSystems/MASM/issues/1622.

## Possible Solution

When working on issue #1625, it was discovered that the `TillageImplement` class was in the EEE module. There may be other data classes in there as well that are shared between multiple modules. Extract `TillageImplement` and any other data classes found that are used by EEE and another module and put them in the `data_classes` directory.

**Comment by ew3361zh** [comment]
Proposed EEE structures to move to general `data_structures` dir:
- `CropSpecies` Enum - used by EEE/emissions.py and multiple places in C&S (maybe will be used by Feed Storage eventually as well) - **edit** @Isgotkowitz says this will be deprecated with the ration formulation refactor so I don't think it needs to be moved.
- `TillageImplement` Enum - used by events.py which is already in RUFAS/data_structures as well as 2 tillage .py files in C&S.

**Comment by Isgotkowitz** [comment]
> * CropSpecies Enum - used by EEE/emissions.py and multiple places in C&S (maybe will be used by Feed Storage eventually as well) - edit @Isgotkowitz says this will be deprecated with the ration formulation refactor so I don't think it needs to be moved.

Yes! The `CropSpecies` enum used to map crops grown in RuFaS to purchased feeds fed to cows. This is necessary in EEE because it has to reconcile the totals fed with the totals of feeds grown, because no on farm feeds are currently fed to the animals. But this will change with the animal refactor, which is currently in progress so I think it would be better to leave that structure alone for now, and it will be removed as part of that refactor.
